keps/sig-windows/20190103-windows-node-support.md

```diff
@@ -110,13 +129,51 @@ As of 29-11-2018 much of the work for enabling Windows nodes has already been co
 
 **Second class support**: Kubernetes contributors are likely to be thinking of Linux-based solutions to problems, as Linux remains the primary OS supported. Keeping Windows support working will be an ongoing burden potentially limiting the pace of development. 
 
-**User experience**: Users today will need to use some combination of taints and node selectors in order to keep Linux and Windows workloads separated. In the best case this imposes a burden only on Windows users, but this is still less than ideal.
+**User experience**: Users today will need to use some combination of taints and node selectors in order to keep Linux and Windows workloads separated. In the best case this imposes a burden only on Windows users, but this is still less than ideal. The recommended approach is outlined below
 
-## Graduation Criteria
+#### Ensuring OS-specific workloads land on appropriate container host
+As you can see below, we plan to document how Windows containers can be scheduled on the appropriate host using Taints and Tolerations. All nodes today have the following default labels
+- beta.kubernetes.io/os = [windows|linux]
+- beta.kubernetes.io/arch = [amd64|arm64|...]
 
+If a deployment does not specify a nodeSelector like `"beta.kubernetes.io/os": windows`, it is possible the Pods can be scheduled on any host, Windows of Linux. This can be problematic since a Windows container can only land on Windows and a Linux container can only land on Linux. The best practice we will recommend is to use a nodeSelector. 
 
-## Implementation History
+However, we understand that in certain cases customers have a pre-existing large number of deployments for Linux containers. Since they will not want to change all deployments to add nodeSelectors, the alternative is to use Taints. Because the kubelet can set Taints during registration, it could easily be modified to automatically add a taint when running on Windows only (`“--register-with-taints=’os=Windows:NoSchedule’” `). By adding a taint to all Windows nodes, nothing will be scheduled on them (that includes existing Linux Pods). In order for a Windows Pod to be scheduled on a Windows node, it would need both the nodeSelector to choose Windows, and a toleration.
+```
+nodeSelector:
+    "beta.kubernetes.io/os": windows
+tolerations:
+    - key: "os"
+      operator: "Equal"
+      Value: “Windows”
+      effect: "NoSchedule"
+```
 
+## Graduation Criteria
+- All features and functionality under `What works today` is fully tested and vetted to be working by SIG-Windows
+- SIG-Windows has high confidence to the stability and reliability of Windows Server containers on Kubernetes
+- 100% green/passing conformance tests that are applicable to Windows (see the Testing Plan section for details on these tests)
+- Comprehensive documentation that includes but is not limited to the following sections
```

Comment:
Is there a plan for where these docs will live?

Reply:
updated to include the location

Reply:
We need to have a plan for where the docs will live, and how they will be written. But finalizing those plans shouldn't be a prerequisite for calling this KEP /implementable